inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/schedulers/ConfigScheduler.kt

```diff
@@ -4,41 +4,45 @@ import androidx.work.ExistingWorkPolicy
 import androidx.work.OneTimeWorkRequest
 import androidx.work.WorkManager
 import com.rakuten.tech.mobile.inappmessaging.runtime.InAppMessaging
+import com.rakuten.tech.mobile.inappmessaging.runtime.utils.RetryDelayUtil
 import com.rakuten.tech.mobile.inappmessaging.runtime.utils.WorkManagerUtil
 import com.rakuten.tech.mobile.inappmessaging.runtime.workmanager.workers.ConfigWorker
+import java.util.concurrent.TimeUnit
 
 /**
  * Scheduling workers to do their work in the background to communicate with IAM config service.
  */
 @SuppressWarnings("PMD.UseUtilityClass")
 internal interface ConfigScheduler {
 
-    fun startConfig()
+    fun startConfig(delay: Long = 0)
 
     companion object {
         private const val CONFIG_WORKER_NAME = "iam_config_worker"
         private var instance: ConfigScheduler = ConfigSchedulerImpl()
+        internal var currDelay = RetryDelayUtil.INITIAL_BACKOFF_DELAY
 
         fun instance() = instance
     }
 
     private class ConfigSchedulerImpl : ConfigScheduler {
-        override fun startConfig() {
+        override fun startConfig(delay: Long) {
             WorkManager.getInstance(InAppMessaging.instance().getHostAppContext()!!)
                     .beginUniqueWork(
                             CONFIG_WORKER_NAME,
                             ExistingWorkPolicy.REPLACE,
-                            getConfigWorkRequest()!!)
+                            getConfigWorkRequest(delay)!!)
                     .enqueue()
         }
 
         /**
          * This method syncs with config service in the background with all necessary input data
          * bundled into Data and passed into Worker.
          */
-        private fun getConfigWorkRequest(): OneTimeWorkRequest? =
+        private fun getConfigWorkRequest(delay: Long): OneTimeWorkRequest? =
                 // Creating a config work request.
                 OneTimeWorkRequest.Builder(ConfigWorker::class.java)
```

Comment:
what about `.setBackoffCriteria`? Do we want to use 10s (for non 429 errors)  just like on iOS?

Reply:
The default setting is used for other errors, which is exponential backoff and initial delay of 30s.

Reply:
Ok so let's leave it for now.